### PR TITLE
[SPMD] Remove _specify_ddp_gpu_num method

### PR DESCRIPTION
**Pull request by wayi1**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56425 [SPMD] Remove _specify_ddp_gpu_num method**
* #55946 [SPMD] Remove ddp_gpu_size field from SyncBatchNorm

As SPMD mode is gone, `_specify_ddp_gpu_num` becomes useless. It only checks if the module is a GPU module. This actually is already checked by the caller of this function (in fairscale and some other codebases).

Facebook:
Additionally also remove `enable_pytorch_sync_bn` wrapper that only calls this function and does nothing else.

Differential Revision: [D27866440](https://our.internmc.facebook.com/intern/diff/D27866440/)

### CODE REVIEW

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -659,12 +658,6 @@ def _check_non_zero_input_channels(self, input):
                 "SyncBatchNorm number of input channels should be non-zero"
             )
 
-    def _specify_ddp_gpu_num(self, gpu_size):
```
Comment (zhaojuanmao) [inline torch/nn/modules/batchnorm.py:662]: was the callee removed from fairscale repo as well?

**Review by zhaojuanmao — APPROVED** [review]

File: torch/nn/modules/batchnorm.py (hunk 1 above)
Comment (SciPioneer) [reply torch/nn/modules/batchnorm.py:662]: Yes, the relevant deletions are shown in the internal diffs, as it's not in PyTorch repo.
